ansible/playbooks/tasks/aws-cluster-bootstrap.yaml

```diff
@@ -0,0 +1,404 @@
+  ansible.builtin.set_fact:
+    max_tasks: infinity
+  loop: "{{ systemd_result.stdout_lines }}"
+  when: item is search("DefaultTasksMax=infinity")
+
+- name: Check max tasks integer
+  ansible.builtin.set_fact:
+    max_tasks_int: "{{ item | split('=') | last }}"
+  loop: "{{ systemd_result.stdout_lines }}"
+  when: item is search("DefaultTasksMax=") and item is not search("DefaultTasksMax=infinity")
+
+- name: Set DefaultTasksMax
+  ansible.builtin.lineinfile:
+    path: /etc/systemd/system.conf
+    regexp: '^DefaultTasksMax='
+    line: 'DefaultTasksMax=4096'
+    state: present
+    backup: true
+  when: max_tasks is defined and max_tasks is not match('infinity')
+
+- name: Set DefaultTasksMax
+  ansible.builtin.lineinfile:
+    path: /etc/systemd/system.conf
+    regexp: '^DefaultTasksMax='
+    line: 'DefaultTasksMax=4096'
+    state: present
+    backup: true
+  when:
+    - max_tasks is defined
+    - max_tasks_int is defined
+    - max_tasks_int | int < 4096
+
+- name: Flush handlers
+  meta: flush_handlers
+
+# check crm status and fail if not return code != 0
+- name: Check the cluster is up
+  ansible.builtin.command:
+    cmd: crm status
+  changed_when: false
+
+- name: Get general cluster configuration
+  ansible.builtin.command:
+    cmd: crm configure show
+  register: crm_conf_show
+  changed_when: false
+
+- name: Get rsc-options cluster configuration
+  ansible.builtin.command:
+    cmd: crm configure show rsc-options
+  register: crm_rsc_options_show
+  changed_when: false
+  failed_when: false
+
+- name: Get op_defaults cluster configuration
+  ansible.builtin.command:
+    cmd: crm configure show op_defaults
+  register: crm_op_options_show
+  changed_when: false
+  failed_when: false
+
+- name: Set cluster facts
+  ansible.builtin.set_fact:
+    sbd_stonith: "{{ (crm_conf_show.stdout | regex_search('(rsc_iscsi_sbd)', '\\1'))[0] | default('false') }}"
+    stonith_enabled: "{{ (crm_conf_show.stdout | regex_search('stonith-enabled=([a-z]*)', '\\1'))[0] | default('false') }}"
+    stonith_action: "{{ (crm_conf_show.stdout | regex_search('stonith-action=([a-z]*)', '\\1'))[0] | default('false') }}"
+    stonith_timeout: "{{ (crm_conf_show.stdout | regex_search('stonith-timeout=([a-z0-9]*)', '\\1'))[0] | default('false') }}"
+    rsc_ip: "{{ crm_conf_show.stdout | regex_search('(rsc_ip_)') }}"
+    rsc_healthcheck_primary: "{{ crm_conf_show.stdout | regex_search('(rsc_healthcheck_primary)') }}"
+    grp_ip_hc: "{{ crm_conf_show.stdout | regex_search('(grp_ip_hc)') }}"
+    resource_stickiness: "{{ (crm_rsc_options_show.stdout | regex_search('resource-stickiness=([0-9]*)', '\\1'))[0] | default('false') }}"
+    migration_threshold: "{{ (crm_rsc_options_show.stdout | regex_search('migration-threshold=([0-9]*)', '\\1'))[0] | default('false') }}"
+    op_default_timeout: "{{ (crm_op_options_show.stdout | regex_search('timeout=([0-9]*)', '\\1'))[0] | default('false') }}"
+
+- name: Stonith command [AWS]
```

Comment:
As far as I can see, this task `Stonith command [AWS]` is only about setting a variable `aws_stonith_cmd`  
This is something has been added https://github.com/SUSE/qe-sap-deployment/pull/257
This is fine.

But `aws_stonith_cmd`  is only used in task named `Configure AWS EC2 STONITH for sle 12sp5` that is just after a task named `Enable SBD [sbd]` that does not use it.

I propose to change order like
- Enable SBD [sbd]
- Stonith command [AWS]
- Configure AWS EC2 STONITH for sle 12sp5

Doing that will keep `Enable SBD [sbd]` in a long continuous section that AWS and GCP has in common, taht start from `Ensure cluster dependencies are installed`

Reply:
Yes, really good thoughts.